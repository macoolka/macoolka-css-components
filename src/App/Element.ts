import { esn, lens } from '../css';
import { content, body, navContent, maskModal } from './style';
export const Page = esn({
    displayName: 'App.Page',
    defaultProps: {},
    tag: 'div',
    style: body,
});
export const AppContent = esn({
    displayName: 'App.Content',
    defaultProps: {},
    tag: 'div',
    style: content,
});
export interface AppNavContentProp {
    expanded: boolean;
}
export const AppNavContent = esn<AppNavContentProp>({
    displayName: 'App.NavContent',
    defaultProps: {
        expanded: false
    },
    tag: 'div',
    style: navContent,
    rule: {
        //expanded: ({ value,source }) => value ? { sideExpand, mkMedia: [{}, {}, { mkZIndex: 'popup' }] } : sideHide,
        expanded: ({ value, theme }) => ({
            paddingBottom: 100,
            marginTop: lens.boxBarHeightLens.get(theme).large,

            paddingLeft: value ? lens.boxContentWidthLens.get(theme).nav : 0,
            boxSizing: 'border-box',
            mkMedia: [{
                marginTop: lens.boxBarHeightLens.get(theme).medium,
                paddingLeft: 0,
                paddingBottom: 50,
            }, {
                paddingLeft: 0,
            }, {
                paddingLeft: 0,
            }],
        })



    },
});

export interface MaskModalProps {
    opened?: boolean;
    navOpened?: boolean;
}
export const MaskModal = esn< MaskModalProps, {}>({
    displayName: 'App.MaskModal',
    defaultProps: { opened: false },
    tag: 'div',
    style: maskModal,
    rule: {
        opened: ({ value }) => value ? ({
            opacity: 1,
            mkVisible: 'visible',
        }) : ({
            mkVisible: 'hidden',
            opacity: .1,
        }),
        navOpened: ({ value }) => value ? ({
            mkMedia: [{

            }, {

            }, {
                mkVisible: 'visible',
                opacity: 1,
            }],
        }) : ({
            mkMedia: [{

            }, {

            }, {
                mkVisible: 'hidden',
                opacity: .1,
            }],
        }),
    },
});
