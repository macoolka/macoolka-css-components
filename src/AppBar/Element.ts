import { esn, lens } from '../css';
import { bar, barWrapper } from './style';
type AppBarProp = {
    full: boolean;
}
export const AppBar=esn<AppBarProp,{},'header'>  ({
    displayName: 'AppBar.AppBar',
    defaultProps: {
        full: true
    },
    tag: 'header',
    style: bar,
    rule: {
        full: ({ value, theme }) => value ? ({ mkPaddingLeft: 'none' }) : ({
            mkTransition: 'padding-left',

            paddingLeft: lens.boxContentWidthLens.get(theme).nav,
            mkMedia: [{
            }, {

            }, {
                mkPaddingLeft: 'none'
            }],
        })
    }
});
export const AppBarWrapper = esn({
    displayName: 'AppBar.Wrapper',
    defaultProps: {},
    tag: 'div',
    style: barWrapper,
});
