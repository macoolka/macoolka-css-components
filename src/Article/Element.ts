import { esn } from '../css';
import {title, titleSmall, article, articleSmall, titleFirst, titleFirstSmall} from './style';
export const Article= esn ({
    displayName: 'Article.Article',
    defaultProps: {
    },
    tag: 'article',
    style: {
        ...article,
        mkMedia: [{
            ...articleSmall,
            selector: {
                ':nth-child(1) > h1,:nth-child(1) > h2': titleFirstSmall,
            },
        }],
        selector: {
            ':nth-child(1) > h1,:nth-child(1) > h2': titleFirst,
        },
    },
});
export const Title=esn ({
    displayName: 'Article.Title',
    defaultProps: {
    },
    tag: 'h2',
    style: {
        ...title,
        mkMedia: [titleSmall],
    },
});
