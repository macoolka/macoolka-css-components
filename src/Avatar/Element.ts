import { esn } from '../css';
import { avatar } from './style';
export interface AvatarProps {
    mkSize?: 'small' | 'medium' | 'large';
}
export const Avatar= esn< {}, AvatarProps>({
    displayName: 'Avatar',
    defaultProps: {
        mkSize: 'medium'
    },
    tag: 'div',
    style: avatar,
    ruleEnum: {
        mkSize: {
            small: {
                width: 36,
                height: 36,
            },
            medium: {
                width: 48,
                height: 48,
            },
            large: {
                width: 64,
                height: 64,
            }
        },
    },
});
export const AvatarImage= esn<{}, AvatarProps,'img'> ({
    displayName: 'Avatar',
    defaultProps: {
        mkSize: 'medium'
    },
    tag: 'img',
    style: avatar,
    ruleEnum: {
        mkSize: {
            small: {
                width: 36,
                height: 36,
            },
            medium: {
                width: 48,
                height: 48,
            },
            large: {
                width: 64,
                height: 64,
            }
        },
    },
});
