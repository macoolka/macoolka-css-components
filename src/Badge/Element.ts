import { esn } from '../css';
import { item, container } from './style';
export const Container = esn({
    displayName: 'Badge.Container',
    defaultProps: {
    },
    tag: 'span',
    style: container,
});
export const Item = esn({
    displayName: 'Badge.Item',
    defaultProps: {
    },
    tag: 'span',
    style: item,
});
