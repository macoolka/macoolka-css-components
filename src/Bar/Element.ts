import { esn } from '../css';
import { topBar, toolBar } from './style';
export interface TopBarProps {
    mkPosition?: 'fixed' | 'absolute' | 'sticky' | 'static';
}

export const TopBar= esn<{}, TopBarProps,'header'> ({
    displayName: 'Bar.TopBar',
    defaultProps: { },
    tag: 'header',
    style: topBar,
    ruleEnum: {
        mkPosition: {
            fixed: {
                mkFixed: 'top',
            },
            absolute: {
                mkAbsolute: 'top',
            },
            sticky: {
                mkSticky: 'top',
            },
            static: {
                position: 'static',
            },
        },
    },
});

export const ToolBar = esn({
    displayName: 'Bar.Toolbar',
    defaultProps: { },
    tag: 'div',
    style: toolBar,
});
