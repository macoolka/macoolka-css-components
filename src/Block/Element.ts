import { esn } from '../css';
import { item, pItem, mItem } from './style';
export const Item = esn({
    displayName: 'Block.Item',
    defaultProps: {
    },
    tag: 'div',
    style: item,
});
export const PItem= esn({
    displayName: 'Block.PItem',
    defaultProps: {
    },
    tag: 'div',
    style: pItem,
});
export const MItem = esn({
    displayName: 'Block.MItem',
    defaultProps: {
    },
    tag: 'div',
    style: mItem,
});
