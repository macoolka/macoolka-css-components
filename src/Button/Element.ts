import { esn } from '../css';
import { button, topButton } from './style';
export const BasicButton= esn({
    displayName: 'BasicButton',
    defaultProps: {
    },
    tag: 'button',
    style: button,
});
export const TopButton = esn({
    displayName: 'BasicButton',
    defaultProps: {
    },
    tag: 'button',
    style: topButton,
});
export const ButtonText = esn({
    displayName: 'BasicButton',
    defaultProps: {
        mkTypography: 'button',
    },
    tag: 'p',
    style: {},
});
export const StyledButton= esn<
{
    mkVariant?: 'flat' | 'normal' | 'raised' | 'outlined',
},{},'button'>({
    displayName: 'StyledButton',
    defaultProps: {
        mkVariant: 'flat',
    },
    tag: 'button',
    style: button,
    ruleEnum: {
        mkVariant: {
            flat: {
                mkColor:'transparent' 
            },
            normal: {
                mkColor:'divide' 
            },
            raised: {
                mkColor:'divide' ,
                mkShadow: 2,
                selector: {
                    ':disabled ': {
                        mkShadow: 0,
                    },
                    ':active:not(:disabled)': {
                        mkShadow: 8,
                    },
                },
            },
            outlined: {
                mkBorder: 'bordered',
                mkColor:'transparent' ,
                mkBorderColor: { name: 'currentColor', opacity: 'large' },
            },
        },
    },
});

export const TextButton=esn< 
{}, {
    mkSize?: 'small' | 'medium' | 'large'
},'button'> ({
    displayName: 'TextButton',
    defaultProps: {
        mkSize: 'medium',
    },
    tag: 'button',
    style: {
        ...button,
        mkTypography: 'button',
        mkBorderRadius: 'small',
    },
    ruleEnum: {
        mkSize: {
            small: ({
                minWidth: 64,
                height: 32,
                mkPaddingH: 'small',
                mkFontSize: 'caption',
            }),
            medium: ({
                minWidth: 88,
                height: 36,
                mkPaddingH: 'medium',
            }),
            large: ({
                minWidth: 112,
                minHeight: 40,
                mkPaddingH: 'large',
                mkFontSize: 'subtitle',
            }),
        },
    },
});
export const IconButton= esn<
{}, {
    mkSize?: 'small' | 'medium' | 'large'
},'button'>  ({
    displayName: 'IconButton',
    defaultProps: {
        mkSize: 'medium',
    },
    tag: 'button',
    style: {
        ...button,
        mkRounded: true,
    },
    ruleEnum: {
        mkSize: {
            small: ({
                mkSquare: 32,

            }),
            medium: ({
                mkSquare: 48,
            }),
            large: ({
                mkSquare: 56,
            }),
        },
    },
});
export const FabButton=esn < { mini?: boolean },{},'button'> ({
    displayName: 'FabButton',
    defaultProps: {
        mini: false,
    },
    tag: 'button',
    style: {
        ...button,
        mkRounded: true,
        width: 56,
        height: 56,
        mkShadow: 6,
        mkColor: 'accent',
        selector: {
            ':active': {
                mkShadow: 12,
            },
        },
    },
    rule: {
        mini: ({ value }: { value: boolean }) => value ? ({
            width: 40,
            height: 40,
        }) : ({
            width: 56,
            height: 56,
        }),
    },
});
export const FabTextButton=esn< { mini?: boolean },{},'button'>  ({
    displayName: 'FabTextButton',
    defaultProps: {
        mini: false,
    },
    tag: 'button',
    style: {
        ...button,
        mkColor: 'accent',
        borderRadius: 48 / 2,
        mkPaddingH: 'large',
        mkPaddingV: 'none',
        width: 'initial',
        minWidth: 48,
        height: 48,
    },
    rule: {
        mini: (value) => value ? ({
            minWidth: 40,
            height: 40,
        }) : ({}),
    },
});
