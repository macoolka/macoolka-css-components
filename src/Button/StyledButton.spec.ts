
import { parse } from '../css';
import { StyledButton } from './Element'
describe('StyledButton', () => {
    it(' StyledButton variant text', () => {
        expect(parse(StyledButton)({
            mkVariant: 'flat',
        })).toMatchSnapshot();
    });
    it(' StyledButton variant text cancel hover color', () => {
        expect(parse(StyledButton)({
            mkVariant: 'flat',
            mkHoverTextColor: 'none',
        })).toMatchSnapshot();
    });
    it(' StyledButton variant container', () => {
        expect(parse(StyledButton)({
            mkVariant: 'raised',
        })).toMatchSnapshot();
    });
    it(' StyledButton variant outline', () => {
        expect(parse(StyledButton)({
            mkVariant: 'outlined',
        })).toMatchSnapshot();
    });
});