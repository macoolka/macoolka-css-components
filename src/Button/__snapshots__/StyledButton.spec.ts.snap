// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton  StyledButton variant container 1`] = `
"border-width: 0px;
overflow-x: hidden;
overflow-y: hidden;
display: inline-flex;
align-items: center;
justify-content: center;
list-style: none;
-webkit-overflow-scrolling: touch;
box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14),
  0px 3px 1px -2px rgba(0, 0, 0, 0.12);
color: rgba(0, 0, 0, 0.87);
background-color: #e0e0e0;
font-family: Roboto, Open Sans, Helvetica Neue, Helvetica, Arial, sans-serif;
text-decoration: none;
user-select: none;
outline: none;
position: relative;
cursor: pointer;
:active:not(:disabled) {
  box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2), 0px 8px 10px 1px rgba(0, 0, 0, 0.14),
    0px 3px 14px 2px rgba(0, 0, 0, 0.12);
}
:disabled {
  color: rgba(0, 0, 0, 0.38);
  pointer-events: none;
  cursor: default;
}
:disabled {
  box-shadow: none;
}
> * {
  margin-left: 8px;
}
> *:first-child {
  margin-left: 0px;
}
"
`;

exports[`StyledButton  StyledButton variant outline 1`] = `
"border-width: 1px;
border-style: solid;
overflow-x: hidden;
overflow-y: hidden;
display: inline-flex;
align-items: center;
justify-content: center;
list-style: none;
-webkit-overflow-scrolling: touch;
color: rgba(0, 0, 0, 0.87);
border-color: rgba(0, 0, 0, 0.47);
background-color: transparent;
font-family: Roboto, Open Sans, Helvetica Neue, Helvetica, Arial, sans-serif;
text-decoration: none;
user-select: none;
outline: none;
position: relative;
cursor: pointer;
:disabled {
  color: rgba(0, 0, 0, 0.38);
  pointer-events: none;
  cursor: default;
}
> * {
  margin-left: 8px;
}
> *:first-child {
  margin-left: 0px;
}
"
`;

exports[`StyledButton  StyledButton variant text 1`] = `
"border-width: 0px;
overflow-x: hidden;
overflow-y: hidden;
display: inline-flex;
align-items: center;
justify-content: center;
list-style: none;
-webkit-overflow-scrolling: touch;
color: rgba(0, 0, 0, 0.87);
background-color: transparent;
font-family: Roboto, Open Sans, Helvetica Neue, Helvetica, Arial, sans-serif;
text-decoration: none;
user-select: none;
outline: none;
position: relative;
cursor: pointer;
:disabled {
  color: rgba(0, 0, 0, 0.38);
  pointer-events: none;
  cursor: default;
}
> * {
  margin-left: 8px;
}
> *:first-child {
  margin-left: 0px;
}
"
`;

exports[`StyledButton  StyledButton variant text cancel hover color 1`] = `
"border-width: 0px;
overflow-x: hidden;
overflow-y: hidden;
display: inline-flex;
align-items: center;
justify-content: center;
list-style: none;
-webkit-overflow-scrolling: touch;
color: rgba(0, 0, 0, 0.87);
background-color: transparent;
font-family: Roboto, Open Sans, Helvetica Neue, Helvetica, Arial, sans-serif;
text-decoration: none;
user-select: none;
outline: none;
position: relative;
cursor: pointer;
:disabled {
  color: rgba(0, 0, 0, 0.38);
  pointer-events: none;
  cursor: default;
}
> * {
  margin-left: 8px;
}
> *:first-child {
  margin-left: 0px;
}
"
`;
